src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -1185,12 +1186,22 @@ protected boolean isCategoryEnabled(@NonNull SCMHeadCategory category) {
      */
     @Override
     public void afterSave() {
-        SCMSourceOwner owner = getOwner();
+        SCMSourceOwner owner = getOwnerNotSynchronized();
         if (owner != null) {
-            GitHubWebHook.get().registerHookFor(owner);
+            getHook().registerHookFor(owner);
         }
     }
 
+    @VisibleForTesting
+    SCMSourceOwner getOwnerNotSynchronized() {
+        return getOwner();
+    }
+
+    @VisibleForTesting
+    GitHubWebHook getHook() {
```

Comment:
This just seems to be ad bad smell if you need to litter `getHook` methods in each class